trick_sims/SIM_aircraft/models/Aircraft/src/Waypoint.cpp

```diff
@@ -0,0 +1,89 @@
+/********************************* TRICK HEADER *******************************
+PURPOSE: ( Handles the waypoints. )
+*******************************************************************************/
+#include "Aircraft/include/Waypoint.hh"
+
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include "Waypoint.hh"
+
+WaypointList::WaypointList() {
+    length = 0;
+}
+
+WaypointList::WaypointList(std::string path) {
+    length = 0;
+
+    load(path);
+}
+
+int WaypointList::add(double n, double w, std::string i)
+{
+    if (length < ARRAY_SIZE)
+    {
+        north[length] = n;
+        west[length] = w;
+        img[length] = i;
+        length++;
+        return length-1;
+    }
+    
+    return -1;
+}
+
+void WaypointList::remove(int index)
+{
+    if (index < 0 || index >= length) {
+        std::cerr << "Invalid Index" << std::endl;
+        return;
+    }
+
+    length--;
+    for(int i = index; i < length; i++) {
+        north[i] = north[i+1];
+        west[i] = west[i+1];
+        img[i] = img[i+1];
+    }
+}
+
+void WaypointList::clear() {
+    length = 0;
+}
+
+void WaypointList::load(std::string path) {
+    clear();
+    append(path);
+}
+
+void WaypointList::append(std::string path) {
+    std::ifstream in(path);
+
+    if(in.is_open()) {
+        std::string str;
+        while(getline(in,str)) {
+            int k = str.find(','), j = str.find(',',k+1);
```

Comment:
I'd add a comment here that says the expected format of what this is parsing, something like
```
// Expected format: <north>,<west>,<img_file>
```

Reply:
This is a great idea. Comment added